signature/algorithm.go

```diff
@@ -0,0 +1,70 @@
+package signature
```

Comment:
> Created a new `signature` submodule to refactor the original `signer` submodule.

It is not clear to me if this is new code or existing code. Can you add the changes you made to `signer` submodule to this PR?

Reply:
Actually, we will do the refactoring in 3 steps. 
1. Create a new `signature` submodule which does the same business as `signer` submodule but introduces proposed refactoring.
2. Add COSE support.
3. Switch to the `signature` submodule and delete the `signer` folder once we test it with `notation` and `notation-go`.